packages/web-pkg/src/components/AppBanner.vue

```diff
@@ -0,0 +1,153 @@
+<template>
+  <portal to="app.app-banner">
+    <div class="app-banner hide-desktop" :hidden="isVisible === false">
+      <oc-button
+        variation="brand"
+        appearance="raw"
+        class="app-banner-exit"
+        aria-label="Close"
+        @click="close"
+      >
+        <oc-icon name="close" size="small" />
+      </oc-button>
+      <div
+        class="app-banner-icon"
+        :style="{ 'background-image': `url('${appBannerSettings.icon}')` }"
+      ></div>
+      <div class="info-container">
+        <div>
+          <div class="app-title">{{ appBannerSettings.title }}</div>
+          <div class="app-publisher">{{ appBannerSettings.publisher }}</div>
+          <div v-if="appBannerSettings.additionalInformation !== ''" class="app-additional-info">
+            {{ $gettext(appBannerSettings.additionalInformation) }}
+          </div>
+        </div>
+      </div>
+      <a
+        :href="appUrl"
+        target="_blank"
+        class="app-banner-cta"
+        rel="noopener"
+        aria-label="{{ $gettext(appBannerSettings.ctaText) }}"
+        >{{ $gettext(appBannerSettings.ctaText) }}</a
+      >
+    </div>
+  </portal>
+</template>
+
+<script lang="ts">
+import { computed, defineComponent, ref, unref } from 'vue'
+import { useRouter, useStore } from 'web-pkg'
+import { buildUrl } from 'web-pkg/src/helpers/router'
+import { useSessionStorage } from '@vueuse/core'
+
+export default defineComponent({
+  components: {},
+  props: {
+    fileId: {
```

Comment:
Not quite sure if this should be always required, as a generic component this could be shown even if not public link / file context